content/tutorial-ios/ios-03.md

```diff
@@ -0,0 +1,144 @@
+
+Let's quickly understand the syntax of the parametrized query. When declaring the query named `Trainer`, we can add as many arguments to it as we like by listing them in parantheses. For each argument, we need to provide a _name_ and a _type_. In our case, we only have one (mandatory) argument called `name` that is of type `String`. Variables are always prepended with a `$`. Then, _inside_ the query we can just access the variable by referencing it with its name. 
+
+Go ahead and replace the existing query in `PokedexTableViewController.graphql` with the new query that accepts the trainer's name as a variable, then build the project again using `CMD + B` and inspect `API.swift`. The only initializer of `TrainerQuery` now takes a `String` as an argument which it assigns to a newly added property called `name`. This is also precisely the reason why the compiler throws an error right now, since in our previous usage of the initializer in `PokedexTableViewController.swift`, we didn't provide any arguments. So, let's go and fix that!
+
+Open `PokedexTableViewController.swift` and add the `name` argument when instantiating `TrainerQuery`:
+
+```swift
+let trainerQuery = TrainerQuery(name: "__NAME__")
+```
+
+Run the app again to validate that everything still works as before.
+
+
+### Nested Queries
+
+Now that we saw query variables in action, we can focus on displaying the Pokemons of a given trainer. We are going to use the second section of our table view for that, each pokemon will be displayed in a `PokemonCell`.
+
+Therefore, we need to be able to ask for all the `ownedPokemons` of a trainer. We can do so by _nesting_ a query for pokemon data inside the query for the trainer. It'll look like this:
+
+```graphql
+query Trainer($name: String!) {
+  Trainer(name: $name) {
+    id
+    name
+    ownedPokemons {
+      id
+      name
+      url
+    }
+  }
+}
+```
+
+Nested queries are where GraphQL really shines compared to REST. With a REST API, it usually requires multiple calls to request data that goes over one or two relationships in a data model, with GraphQL all data requirements can be specified upfront in the query and the data can be fetched within only one API call.
+
+The above query requests all Pokemons that are owned by the trainer called `$name`. The Pokemons will be returned as an array, and each of them will have an `id`, a `name` and a `url`.
+
+Go ahead and replace the existing query in `PokedexTableViewController.graphql` with the query above, then hit `CMD + B` and inspect `API.swift` again. 
+
+We just got another nested struct, this time it's called `OwnedPokemon` and nested inside `TrainerQuery.Data.Trainer`. This well reflects the nesting inside the query. Also notice that the `TrainerQuery.Data.Trainer` got a new property, which is an array of `TrainerQuery.Data.Trainer.OwnedPokemon`.
```

Comment:
This well reflects the nesting inside the query. 
->
This reflects the nesting inside the query.